src/mscorlib/src/System/Collections/Generic/List.cs

```diff
@@ -247,7 +246,7 @@ public void Add(T item)
             var array = _items;
             var size = _size;
             _version++;
-            if ((uint)size < (uint)array.Length)
+            if (array.IndexInRange(size))
```

Comment:
The issue is negative numbers; what's the cost of `checked`? Could morph array access to cast to `uint` and catch the overflow? Or will that be the same as doing the bounds check?